src/test/java/uk/gov/ons/ssdc/caseprocessor/messaging/ManagedMessageRecovererTest.java

```diff
@@ -0,0 +1,200 @@
+package uk.gov.ons.ssdc.caseprocessor.messaging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.cloud.gcp.pubsub.support.BasicAcknowledgeablePubsubMessage;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.MessagingException;
+import org.springframework.retry.RetryContext;
+import uk.gov.ons.ssdc.caseprocessor.client.ExceptionManagerClient;
+import uk.gov.ons.ssdc.caseprocessor.model.dto.ExceptionReportResponse;
+import uk.gov.ons.ssdc.caseprocessor.model.dto.SkippedMessage;
+
+@ExtendWith(MockitoExtension.class)
+class ManagedMessageRecovererTest {
+  @Mock private BasicAcknowledgeablePubsubMessage originalMessage;
+
+  @Mock private ExceptionManagerClient exceptionManagerClient;
+
+  @InjectMocks private ManagedMessageRecoverer underTest;
+
+  @Test
+  public void testRecover() throws Exception {
+    // Given
+    ExceptionReportResponse exceptionReportResponse = new ExceptionReportResponse();
+    RetryContext retryContext = testSetupTestRecover(exceptionReportResponse);
+
+    // When
+    underTest.recover(retryContext);
+
+    // Then
+    verify(exceptionManagerClient)
+        .reportException(
+            eq("90f56b5b3ffe9558a546af25a7256da4b2761864575f9d59c81b70629023465b"),
```

Comment:
Lots of copy paste of this same hash string 🎨

Reply:
fixed